api/api/controllers/search_controller.py

```diff
@@ -191,13 +192,13 @@ def get_excluded_providers_query() -> Q | None:
                 key=FILTERED_PROVIDERS_CACHE_KEY,
                 version=FILTERED_PROVIDERS_CACHE_VERSION,
                 timeout=FILTER_CACHE_TIMEOUT,
-                value=filtered_providers,
+                value=enabled_providers,
             )
         except ConnectionError:
             logger.warning("Redis connect failed, cannot cache filtered providers.")
 
-    if filtered_providers:
-        return Q("terms", provider=filtered_providers)
+    if enabled_providers:
+        return Q("terms", provider=enabled_providers)
```

Comment:
That's along the lines of what I'm thinking... but _if and only if_ all active sources have a `ContentProvider` entry.

Which is to say, _exclusion_ was certainly safer for sources (though the code was already problematic in that we couldn't disable a source).

Clarifying whether `ContentProvider` is _just providers_ or if it's also sources (and if not, then how _sources_ factor into the data model) would be really helpful.

If it's just providers, then this code would work fine, if all we want is to _include providers that have a `ContentProvider` and are not disabled_. However, if we also need to be able to selectively enable/disable _sources_, then either: (a) we need to overload the conception of `ContentProvider`, and probably stick to only explicitly _excluding_ sources rather than including them based on `ContentProvider` (otherwise we'd have to make a `ContentProvider` for all the Smithsonian sources); or, (b) we need to create a separate `Source` model to track sources, and then probably only track source _exclusions_, considering how many sources there are and how tedious it would be to manage that by hand... though that tediousness is maybe just this first go-round and we could automatically create sources during the data refresh when a new one is detected.

All to say: this is pretty complicated _but_ so long as we are not currently _relying on `ContentProvider` to exclude a **source**_  then it should work. If that's the case, then great, we can move forward with the PR as is, and then circle back to this question of how to (and whether to) manage source inclusion/exclusion. Otherwise, we probably need to stick to exclusion.

We definitely have sources in `ContentProvider` (finnish_satakunnan_museum and finnish_heritage_agency, the Smithsonian sources, others too), but I can't at a glance tell which are excluded (need to do a db query for that).

These are the only hidden ones:

```
deploy@localhost:openledger> select provider_name from content_provider where filter_content;
+---------------------+
| provider_name       |
|---------------------|
| Flora-On            |
| ccMixter            |
| Science Museum – UK |
+---------------------+
SELECT 3
Time: 0.222s
```

All of these are "provider-sources", so it should be okay _for now_. But if we make this change to the code, we really need to circle back and reconsider the meaning of `ContentProvider` and the provider/source relationship in that model.

It would probably be a good idea to have separate models, `Provider` and `Source`, where `Source` has a foreign key relationship to `Provider`, rather than overloading the meaning of `ContentProvider`. Even just reasoning through the limitations of the current approach to query building and the meaning of these models would be so much easier if that distinction was clear in absolute terms.

Reply:
Originally, we only had `source`. Then, the `provider` field was added to denote the provider of the metadata about the image [^1]. The PR that added the `provider` field, didn't touch the `ContentProvider` model, and this model wasn't updated later.

`ContentProvider` currently only handles _sources_, not _providers_. You can check this if you go to https://openverse.org/sources - you'll see NASA in the table. This uses the `stats` endpoint that queries `sources`, not `providers`.

All this is to say that I think we _can_ proceed with this PR, and exactly because we rely on `ContentProvider` to filter sources. I'll review the code again to check that this is true.

> All to say: this is pretty complicated but so long as we are not currently relying on ContentProvider to exclude a source then it should work. If that's the case, then great, we can move forward with the PR as is, and then circle back to this question of how to (and whether to) manage source inclusion/exclusion. Otherwise, we probably need to stick to exclusion.

Could we open a new issue (maybe even a Project proposal request?) to 
- Rename the `ContentProvider` model to `ContentSource`
- Create a new `ContentProvider` model for providers
- surface provider stats in the API
- use this view to add provider links to the frontend single result page
There is some additional info in the cc-archive issues and PRs (https://github.com/cc-archive/cccatalog-api/pull/548/files#r444933371, https://github.com/cc-archive/cccatalog-api/pull/560#issue-654914001)

By the way, some of the issues mention that a source can appear in several providers. I think we have a project for implementing this, this would reduce the number of duplicates we have, but we would need to find a way to handle a work that has several providers. 

[^1]: https://github.com/cc-archive/cccatalog-api/issues/531

Reply:
> All this is to say that I think we can proceed with this PR, and exactly because we rely on ContentProvider to filter sources. I'll review the code again to check that this is true.

If `ContentProvider` is _only_ for sources then we need to change the filter here to `source`, not `provider`.